packages/remix-dev/cli/create.ts

```diff
@@ -518,10 +523,27 @@ export async function validateTemplate(input: string) {
     }
     case "repo": {
       let spinner = ora("Validating the template repo…").start();
-      let { url, filePath } = getRepoInfo(input);
+      let { branch, filePath, owner, name } = getRepoInfo(input);
       let response;
+      let apiUrl = `https://api.github.com/repos/${owner}/${name}`;
+      let method = "HEAD";
+      if (branch) {
+        apiUrl += `/git/trees/${branch}?recursive=1`;
+      }
+      if (filePath) {
+        // When filePath is present, we need to examine the response json to see
+        // if that path exists in the repo.
+        invariant(
+          branch,
+          "expected branch to be present, since the github URL structure requires it (and we need the recursive tree of files to validate)"
```

Comment:
I think we should rephrase this as a less-internal explanation for the end user:
"Expecting branch to be present when specifying a path."